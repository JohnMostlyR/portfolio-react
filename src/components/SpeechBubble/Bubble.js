import styled from 'styled-components';
import {svgToURL} from '../../styles/tools';
import {baseFontStackSerif} from '../../styles/templates/typography';

const Bubble = styled.div`
  position: relative;
  background-color: #F90;
  border-radius: .5rem;
  font-family: ${baseFontStackSerif}
  padding: .5em;

  @media (min-height: 750px) {
    margin-right: 0;
    border-bottom-left-radius: ${props => (props.isLeftHanded) ? 0 : '.5rem'};
    border-bottom-right-radius: ${props => (props.isLeftHanded) ? '.5rem' : 0};

    &:after {
      content: "";
      height: 9vh;
      left: 0;
      position: absolute;
      right: 0;
      top: 100%;
      transform: ${props => (props.isLeftHanded) ? 'rotateY(0)' : 'rotateY(180deg)'};
      background-image: url(${svgToURL('<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 427.9 175.1 132.1" preserveAspectRatio="xMinYMid"><path stroke="#FFAD33" fill="#FFAD33" d="M0 427.9l175.1 65.7-43.7-65.7z"/><path fill="#A36200" d="M175.1 493.6l-87.5 21.9v-54.8"/><path fill="#FFAD33" d="M87.6 515.5l21.2 44.5v-49.8"/></svg>')});
      background-repeat: no-repeat;
      background-size: contain;
      background-position: left top;
    }
  }
`;

export default Bubble;
