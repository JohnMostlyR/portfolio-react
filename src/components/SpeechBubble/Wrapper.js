import styled from 'styled-components';
import {em} from 'polished';
import mq from '../../styles/templates/mediaQueries';
import typography from '../../styles/templates/typography';

import Bubble from './Bubble';

const Wrapper = styled.div`
  margin-bottom: 9vh;
  padding: 1vh 1vw;
  width: 100%;
  ${typography.doublePica};

  ${mq.xl`max-width: ${em('1008px')};`}

  ${Bubble} p {
    &:not(:first-child) {
      margin-top: 1.5rem;
    }
  }
`;

export default Wrapper;
