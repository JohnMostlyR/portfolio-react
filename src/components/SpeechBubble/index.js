import React from 'react';
import PropTypes from 'prop-types';

import Bubble from './Bubble';
import Wrapper from './Wrapper'

const SpeechBubble = (props) => (
    <Wrapper>
      <Bubble isLeftHanded={props.isLeftHanded}>
        {props.children}
      </Bubble>
    </Wrapper>
);

SpeechBubble.propTypes = {
  isLeftHanded: PropTypes.bool,
};

export default SpeechBubble;
