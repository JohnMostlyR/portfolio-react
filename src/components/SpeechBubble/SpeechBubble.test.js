import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {shallow} from 'enzyme';
import SpeechBubble from './index';

describe('SpeechBubble', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
        <SpeechBubble/>,
    );
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(<SpeechBubble/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});