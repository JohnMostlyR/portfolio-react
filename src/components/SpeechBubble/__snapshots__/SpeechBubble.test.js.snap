// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeechBubble Should render and match the snapshot 1`] = `
.c2 {
  position: relative;
  background-color: #F90;
  border-radius: .5rem;
  font-family: "Skolar Regular","Roboto Slab","Droid Serif",Cambria,Georgia,"Times New Roman",Times,serif;
  padding: .5em;
}

.c0 {
  margin-bottom: 9vh;
  padding: 1vh 1vw;
  width: 100%;
  font-size: 1.25rem;
  line-height: 1.5rem;
}

.c0 .c1 p:not(:first-child) {
  margin-top: 1.5rem;
}

@media (min-height:750px) {
  .c2 {
    margin-right: 0;
    border-bottom-left-radius: .5rem;
    border-bottom-right-radius: 0;
  }

  .c2:after {
    content: "";
    height: 9vh;
    left: 0;
    position: absolute;
    right: 0;
    top: 100%;
    -webkit-transform: rotateY(180deg);
    -ms-transform: rotateY(180deg);
    transform: rotateY(180deg);
    background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 427.9 175.1 132.1' preserveAspectRatio='xMinYMid'%3E%3Cpath stroke='%23FFAD33' fill='%23FFAD33' d='M0 427.9l175.1 65.7-43.7-65.7z'/%3E%3Cpath fill='%23A36200' d='M175.1 493.6l-87.5 21.9v-54.8'/%3E%3Cpath fill='%23FFAD33' d='M87.6 515.5l21.2 44.5v-49.8'/%3E%3C/svg%3E");
    background-repeat: no-repeat;
    background-size: contain;
    background-position: left top;
  }
}

@media (min-width:20em) {
  .c0 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 1.625rem;
    line-height: 2.5rem;
  }
}

@media (min-width:63em) {
  .c0 {
    max-width: 63em;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
</div>
`;
